### PR TITLE
examples/bindist: some improvements

### PR DESCRIPTION
**Pull request by kaspar030**
This PR improves some things in our current "bindist" example:

- add link to background information
- add BIN_USEMODULE variable, making selection of module .a redundant
- don't automatically add "$APPLICATION.a", it might not be proprietary
- fix git commands (use "-C" instead of "--workdir")

### CODE REVIEW

File: examples/bindist/Makefile [hunk 1]
```diff
@@ -18,13 +18,14 @@ QUIET ?= 1
...
 # build and use module "abc".
-# use BINARY_DIRS instead of DIRS
-BINARY_DIRS += abc
-USEMODULE += abc
+# use BIN_DIRS instead of DIRS, BIN_USEMODULE instead of USEMODULE,
+# to flag that the directory should only be included when doing a normal build,
+# and that the resulting .a should be saved when doing "make bindist"
+BIN_DIRS += abc
+BIN_USEMODULE += abc        # include "abc" module
+BIN_USEMODULE += bindist    # include application module (named $APPLICATION)
```
Comment (cgundogan) [inline examples/bindist/Makefile]: why is this line necessary? It seems to also work without this line. Could you elaborate on its function?

Comment (cgundogan) [inline examples/bindist/Makefile]: ping @kaspar030 ?

Comment (kaspar030) [reply examples/bindist/Makefile]: This selects the module "bindist", which in this case is the application, to be included in the binary distribution.

**Comment by kaspar030** [comment]
- rebased

**Comment by haukepetersen** [comment]
@cgundogan are you also happy?

**Comment by cgundogan** [comment]
yep ACK and GO
